src/main/scala/org/broadinstitute/dsde/workbench/sam/api/SecurityDirectives.scala

```diff
@@ -51,6 +51,33 @@ trait SecurityDirectives {
       }
     }
 
+  def requireCreateWithParent(maybeParent: Option[FullyQualifiedResourceId], resourceType: ResourceType, userId: WorkbenchUserId, samRequestContext: SamRequestContext): Directive0 = {
```

Comment:
From the name of the method, I would expect this to require a parent to be passed in, but it seems to work without one specified?

Reply:
the calling code looks better if this handles parent both cases (with and without parent) in 1 function... this could be called `maybeRequireCreateWithParent` but that did not make me feel good either. `requireCreateWithParentIfSpecified`?

Reply:
How about `requireCreateWithOptionalParent`